p2p/block_sync.go

```diff
@@ -0,0 +1,98 @@
+package p2p
+
+import (
+	"context"
+
+	"github.com/dymensionxyz/dymint/types"
+	"github.com/ipfs/go-datastore"
+	dsync "github.com/ipfs/go-datastore/sync"
+
+	mh "github.com/multiformats/go-multihash"
+
+	"github.com/ipfs/boxo/bitswap/client"
+	"github.com/ipfs/boxo/bitswap/network"
+	"github.com/ipfs/boxo/bitswap/server"
+	"github.com/ipfs/boxo/blockservice"
+	"github.com/ipfs/boxo/blockstore"
+	"github.com/ipfs/go-cid"
+	routinghelpers "github.com/libp2p/go-libp2p-routing-helpers"
+	"github.com/libp2p/go-libp2p/core/host"
+)
+
+type BlockSync struct {
+	// service that reads/writes blocks either from local datastore or the network
+	bsrv blockservice.BlockService
+	// local datastore for IPFS blocks
+	bstore blockstore.Blockstore
+	// protocol used to obtain blocks from the P2P network
+	net network.BitSwapNetwork
+	// used to retrieve/create merkle DAGs using block data
+	dsrv BlockSyncDagService
+	// used to define content identifiers
+	cidBuilder cid.Builder
+	logger     types.Logger
+
+	// receives blocks obtained from the network
+	msgHandler BlockSyncMessageHandler
+}
+
+type BlockSyncMessageHandler func(block *P2PBlock)
+
+func StartBlockSync(ctx context.Context, h host.Host, store datastore.Datastore, msgHandler BlockSyncMessageHandler, logger types.Logger) (*BlockSync, error) {
+	ds := dsync.MutexWrap(store)
```

Comment:
never returns an error

Reply:
true, i refactored and renamed func to SetuptBlockSync, which makes more sense